### PR TITLE
Target shift

### PR DESCRIPTION
**Pull request by cmougan**
I modified the folders a bit to organize the experiments. 

**"Target shift" Experiment.**

1. Sort the data by target.
2. Split the dataset in 1/3, 2/3, 3/3
3. Train the model in the 2/3
4. Predict uncertainty for the whole model.

Expected results: 
In the image, Linear_Power_Plant_full.png, the splitting points of the data are the lines.

We should see that below the first line there is a increase in uncertainty and above the second one there is a increase in uncertainty
QuantileRegressor_PowerPlant_full

### CODE REVIEW

File: experiments/targetShift.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        X["target"] = y
+        X = X.sort_values("target").reset_index(drop=True)
+
+        # Determine fraction to use in test (will be x2 the split point)
+        split_point = int(X.shape[0] * 0.35)
+
+        # Select smaller and bigger than splitting point
+        train = X.iloc[split_point : X.shape[0] - split_point]
+        pre = X.iloc[:split_point]
+        post = X.iloc[split_point:]
```
Comment (saattrupdan) [inline experiments/targetShift.py:37]: Shouldn't this be `X.shape[0] - split_point` instead of `split_point`?

**Review by saattrupdan — CHANGES_REQUESTED** [review]

File: experiments/targetShift.py (hunk 1 above)
Comment (cmougan) [reply experiments/targetShift.py:37]: Let me think...
Let say the dataset has 300 rows.

I want to select for training the rows 100-200
So I am doing 
[100: 300 - 100]

No? Might be wrong

Comment (saattrupdan) [reply experiments/targetShift.py:37]: Yep, I understand what you're saying, but right now you're defining `pre` to be the first 100 and `post` to be everything after 100, i.e. the interval [100,..., 300). Instead, you want to set `post = X.iloc[X.shape[0] - split_point : ]`.

Comment (cmougan) [reply experiments/targetShift.py]: You are right! Just corrected that!

Comment (cmougan) [reply experiments/targetShift.py]: QuantileRegressor_PowerPlant_full

Comment (cmougan) [reply experiments/targetShift.py]: results look a bit better now

**Pull request merged**